test/e2e/bgppolicy_test.go

```diff
@@ -123,84 +114,177 @@ func TestBGPPolicy(t *testing.T) {
+		notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
+		checkFRRRouterBGPRoutes(t, expectedRoutes, notExpectedRoutes)
+
+		t.Log("Verifying the connectivity of the installed routes on the remote FRR route")
+		ipsToConnect = []string{clusterIP}
+		for _, ip := range ipsToConnect {
+			checkFRRRouterBGPRouteConnectivity(ip)
+		}
+	})
+
+	t.Run("Multiple BGPPolicies applied to different Nodes within a confederation", func(t *testing.T) {
+		t.Log("Configuring the remote FRR router with BGP")
+		configureExternalBGPRouter(t, int32(65000), int32(64512), true)
+
+		asnStart := int32(60000)
+		var bgpPolicies []*crdv1alpha1.BGPPolicy
+		for i := 0; i < len(clusterInfo.nodes); i++ {
+			bgpPolicyName := "test-policy-" + strconv.Itoa(i)
+			localASN := asnStart + int32(i)
+			bgpPolicy := &crdv1alpha1.BGPPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: bgpPolicyName,
+				},
+				Spec: crdv1alpha1.BGPPolicySpec{
+					NodeSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							labelNodeHostname: clusterInfo.nodes[i].name,
+						},
+					},
+					LocalASN:   localASN,
+					ListenPort: ptr.To[int32](179),
+					Confederation: &crdv1alpha1.Confederation{
+						Identifier: int32(64512),
+					},
+					Advertisements: crdv1alpha1.Advertisements{
+						Service: &crdv1alpha1.ServiceAdvertisement{
+							IPTypes: []crdv1alpha1.ServiceIPType{crdv1alpha1.ServiceIPTypeClusterIP},
+						},
+						Pod: &crdv1alpha1.PodAdvertisement{},
+					},
+					BGPPeers: []crdv1alpha1.BGPPeer{
+						{Address: externalInfo.externalFRRIPv4, ASN: int32(65000)},
+					},
+				},
+			}
+			t.Logf("Creating a test BGPPolicy %q with ASN %d, the confederation ID 65000, applied to Node %s", bgpPolicy.Name, localASN, clusterInfo.nodes[i].name)
+			bgpPolicy, err = data.CRDClient.CrdV1alpha1().BGPPolicies().Create(context.TODO(), bgpPolicy, metav1.CreateOptions{})
+			require.NoError(t, err)
+			defer data.CRDClient.CrdV1alpha1().BGPPolicies().Delete(context.TODO(), bgpPolicyName, metav1.DeleteOptions{})
+			bgpPolicies = append(bgpPolicies, bgpPolicy)
+		}
+
+		t.Log("Getting the routes installed on the remote FRR router and verifying them")
+		expectedRoutes := make([]FRRRoute, 0)
+		for _, node := range clusterInfo.nodes {
+			expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: node.podV4NetworkCIDR, Nexthops: []string{node.ipv4Addr}})
+		}
+		expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()})
+		checkFRRRouterBGPRoutes(t, expectedRoutes, nil)
+
+		t.Log("Verifying the connectivity of the installed routes on the remote FRR route")
+		ipsToConnect := []string{podIP, clusterIP}
+		for _, ip := range ipsToConnect {
+			checkFRRRouterBGPRouteConnectivity(ip)
+		}
+
+		t.Log("Updating the BGP configuration on the remote FRR router")
+		configureExternalBGPRouter(t, int32(65000), int32(64513), false)
```

Comment:
maybe you could define `confederationID1 := int32(64512)` and `confederationID2 := int32(64513)` at the beginning of the subtest for clarity?

Reply:
Done